lib/services/internalEngine/index.js

```diff
@@ -488,12 +488,13 @@ class InternalEngine {
    *
    * @param {string} type - data collection
    * @param {object} mapping
+   * @param {string} index - data index (default to this.index)
    * @return {Promise}
    */
-  updateMapping (type, mapping) {
+  updateMapping (type, mapping, index) {
```

Comment:
(Optional) I think the following is more elegant:

```js
updateMapping(type, mapping, index = this.index) {
  const request = {
    type,
    index,
    body: mapping
  };
```

Reply:
I had seen that there were a difference in performance with default arguments but apparently this is not valid with NodeJS (I run some benchmark).

Reply:
I didn't think of that :sweat_smile: 
Good thing you tested it :+1: